force-app/main/service/AddressService.cls

```diff
@@ -151,8 +155,18 @@ public inherited sharing class AddressService {
         sobjDst.put(strFieldPrefix + 'PostalCode', addr.MailingPostalCode__c);
         sobjDst.put(strFieldPrefix + 'Latitude', addr.Geolocation__Latitude__s);
         sobjDst.put(strFieldPrefix + 'Longitude', addr.Geolocation__Longitude__s);
-        sobjDst.put(strFieldPrefix + 'State', addr.MailingState__c);
-        sobjDst.put(strFieldPrefix + 'Country', addr.MailingCountry__c);
+        String countryField = 'Country';
+        String stateField = 'State';
+        if (orgConfig.isStateCountryPicklistsEnabled()) {
+            if (addr.MailingCountry__c != null && addr.MailingCountry__c.length() <= 3) {
```

Comment:
Out of curiosity, why `<= 3`? I'm pretty sure country codes are always 2 characters ([link](https://help.salesforce.com/s/articleView?id=sf.admin_state_country_picklists_standard_countries.htm&type=5))

Reply:
Those are defaults, but 3 character codes also exist: [List of ISO 3166 country codes](https://en.wikipedia.org/wiki/List_of_ISO_3166_country_codes)